### ISSUE TITLE
need to specify C++14 even on UNIX

### ISSUE THREAD

**Issue by mitchblank** [cassioneri/eaf#1]
Two related problems actually:

* setting `CXX_STANDARD` via `GLOBAL PROPERTY` doesn't seem to work on modern cmake versions(?)
* the latest versions of Boost.Math warn about an impending shift to C++14, which breaks the compile due to `-Werror`:
```c
#    warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)"
```

The easiest fix is to just consider C++14 the new minimum version everywhere, and specify it using the the `CMAKE_CXX_STANDARD` setting (see https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) instead of via properties:
```diff
index 954c62e..9d6f4ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,12 +16,11 @@ project(EAF VERSION 1.0
   LANGUAGES CXX
 )
 
+set(CMAKE_CXX_STANDARD 14)
 if (MSVC)
-  set_property(GLOBAL PROPERTY CXX_STANDARD 14)
   string(REGEX REPLACE "/W[0-4]" "/W4 /WX" CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS}")
 else()
-  set_property(GLOBAL PROPERTY CXX_STANDARD 11)
   add_compile_options(-Wall -Wextra -Wpedantic -Werror)
 endif()
```

I think the recommended way of tweaking it on "modern" cmake is via the target-specific "features", i.e. for each executable you'd have...
```cmake
target_link_libraries(to_date benchmark benchmark_main)
target_compile_features(to_date PRIVATE cxx_std_14)
```
...but for this simple use-case probably just setting a global `CMAKE_CXX_STANDARD` is good enough.

**Comment by mitchblank** [comment]
Works for me after the d56da9a commit

**Comment by cassioneri** [comment]
@mitchblank Thanks for reporting. As you've noticed, this has been fixed by commit d56da9abbc8c120b139b4dfac41149923909c5c0.

Here are some comments:

1. I'm not an expert on CMake and I followed your advise on setting `CMAKE_CXX_STANDARD`. (Thank you.)
2. I've globally updated to C++14, i.e., even on Linux.
3. I've upgraded to boost 1.82.0.
4. I've updated benchmark to v1.71.0 because due to a but in v1.7.0 (the latest at the time that I created this repository) the build was set to use a specific commit of benchmark that had the fix. This commit is contained in release v1.7.1.

Despite the updates above, I wish to mention that going forward, I don't plan to keep upgrading dependencies or the required standard. The point of this repository is to provide a reference implementation for the algorithms in the paper and by freezing dependencies and the code at some specific point in time, I hope to maximise the chances of the paper's results to be reproducible on the long term. Keeping upgrading to the latest versions can be risk (things might break) and time consuming.

Thanks again for reporting and for your interest in our work.
